salt/metalk8s/products/init.sls

```diff
@@ -0,0 +1,81 @@
+{% import_yaml "/etc/metalk8s/bootstrap.yaml" as bootstrap_config %}
+
+{% set products = bootstrap_config.products.metalk8s %}
+
+{% set base = '/srv/scality' %}
+{% set roots_file = '/etc/salt/master.d/99-metalk8s-roots.conf' %}
+
+{% for product in products %}
+{%   if salt['file.file_exists'](product) %}
+{%     set product_version = salt['metalk8s.product_version_from_iso'](product) %}
+{%     if not product_version %}
+{%         set product_path = product %}
+{%         set product_version = 'unknown' %}
+{%         set product_path_available = "- test: Product " ~ product ~ " available" %}
+Product {{ product }} available:
+  test.configurable_test_state:
+  - changes: false
+  - result: false
+  - comment: >
+      Unable to retrieve product version from given product file {{ product }},
+      assumed to be an ISO archive containing a "product.txt" file.
+
+{%     else %}
+{%         set product_path = base ~ '/metalk8s-' ~ product_version %}
+{%         set product_path_available = "- mount: Product " ~ product ~ " available at " ~ product_path %}
+Product path {{ product_path }} exists:
+  file.directory:
+  - name: {{ product_path }}
+  - mode: 0755
+  - makedirs: true
+
+Product {{ product }} available at {{ product_path }}:
+  mount.mounted:
+  - name: {{ product_path }}
+  - device: {{ product }}
+  - fstype: iso9660
+  - mkmnt: false
+  - opts: ro
+  - require:
+    - file: Product path {{ product_path }} exists
+{%     endif %} {# product_version is not None #}
+{%   else %}
+{%     set product_version = salt['metalk8s.product_version_from_tree'](product) %}
+{%     set product_path = product %}
+{%     set product_path_available = "- file: Product " ~ product ~ " available" %}
+Product {{ product }} available:
+  file.exists:
+  - name: {{ product_path }}
+{%   endif %}
+
+Assert '{{ product_path }}/product.txt' exists:
+  file.exists:
+  - name: {{ product_path }}/product.txt
+  - require:
+    {{ product_path_available }}
+
+Assert product '{{ product }}' is MetalK8s:
+  cmd.run:
+  - name: grep '^NAME=MetalK8s$' {{ product_path }}/product.txt > /dev/null
+  - stateful: true
+  - require:
+    - file: Assert '{{ product_path }}/product.txt' exists
+
+{%   for (root, subdir) in [('file_roots', 'salt'), ('pillar_roots', 'pillar')] %}
+Expose Salt '{{ root }}' for {{ product_path }}:
```

Comment:
We should directly call the salt-master state instead of doing it here

Reply:
Ok, will do